### PR TITLE
Bento 1214

### PR DESCRIPTION
**Pull request by wfy1997**
## Description

Fix the issue that FileCentricCart navigates back to the first page due to the removal of a record on another page.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

### CODE REVIEW

File: src/components/serverPaginatedTable/serverPaginatedTable.js [hunk 1]
```diff
@@ -29,6 +29,14 @@ class ServerPaginatedTableView extends React.Component {
         direction: this.props.defaultSortDirection,
       },
     });
+    if (this.props.updateSortOrder) {
+      if (localStorage.getItem('rowsPerPage') !== null) {
```
Comment (AjayDoddapaneni) [inline src/components/serverPaginatedTable/serverPaginatedTable.js:33]: Cann you make local storage variable dynamic from the props

File: src/components/serverPaginatedTable/serverPaginatedTable.js [hunk 2]
```diff
@@ -188,8 +245,11 @@ class ServerPaginatedTableView extends React.Component {
 
   render() {
     const {
-      data, count, isLoading, rowsPerPage, sortOrder, className,
+      data, count, isLoading, sortOrder, className,
     } = this.state;
+    const stateRowsPerPage = this.state.rowsPerPage;
+    const rowsPerPage = stateRowsPerPage;
```
Comment (AjayDoddapaneni) [inline src/components/serverPaginatedTable/serverPaginatedTable.js:251]: const rowsPerPage = this.state.rowsPerPage;
this will do the trick?

Comment (wfy1997) [reply src/components/serverPaginatedTable/serverPaginatedTable.js:251]: This is part of the old code that I did not update completely, and I will update it now.

File: src/components/serverPaginatedTable/serverPaginatedTable.js [hunk 3]
```diff
@@ -169,26 +204,50 @@ class ServerPaginatedTableView extends React.Component {
   async fetchData(offset, rowsRequired, sortOrder = {}) {
     let sortDirection = 'asc';
     let sortColumn = 'arm';
-
+    let offsetReal = offset;
+    let page = offset / rowsRequired;
+    // if the offset value is bigger that the count, then change offset value
+    if (offset >= this.props.count) {
+      if (this.props.count % rowsRequired !== 0) {
```
Comment (AjayDoddapaneni) [inline src/components/serverPaginatedTable/serverPaginatedTable.js:211]: page = this.props.count % rowsRequired !== 0 ? Math.floor(this.props.count / rowsRequired) :Math.floor(this.props.count / rowsRequired) - 1;
much cleaner way

File: src/pages/fileCentricCart/cartController.js [hunk 4]
```diff
@@ -9,20 +9,37 @@ import CartView from './cartView';
 const cartController = () => {
   const cart = getCart();
   const ids = cart.fileIds ? cart.fileIds : [];
-  const defaultSortDirection = cart.sortDirection === '' ? table.defaultSortDirection || 'asc' : cart.sortDirection;
+  const defaultSortDirection = cart.sortDirection === '' || !cart.sortDirection || cart.sortDirection === null ? table.defaultSortDirection || 'asc' : cart.sortDirection;
   const CART_QUERY = defaultSortDirection === 'desc' ? GET_MY_CART_DATA_QUERY_DESC : GET_MY_CART_DATA_QUERY;
-  const defaultSortCoulmnValue = cart.sortColumn === '' ? table.defaultSortField || '' : cart.sortColumn;
+  const defaultSortColumnValue = cart.sortColumn === '' || !cart.sortColumn || cart.sortColumn === null ? table.defaultSortField || '' : cart.sortColumn;
```
Comment (AjayDoddapaneni) [inline src/pages/fileCentricCart/cartController.js:14]: This looks like lot of clumsy code can we refactor this.

File: src/components/serverPaginatedTable/serverPaginatedTable.js (hunk 3 above)
Comment (wfy1997) [reply src/components/serverPaginatedTable/serverPaginatedTable.js:211]: let me try

File: src/pages/fileCentricCart/cartController.js (hunk 4 above)
Comment (wfy1997) [reply src/pages/fileCentricCart/cartController.js:14]: Make this way to make sure even if the user is using the page for the first time, we still have the default value for the filecentric table.

**Review by AjayDoddapaneni — APPROVED** [review]

**Pull request merged**